src/Pretzel.Logic/Templating/Context/Data.cs

```diff
@@ -0,0 +1,203 @@
+using System.Collections.Generic;
+using DotLiquid;
+using Pretzel.Logic.Extensions;
+using System.IO;
+using System.IO.Abstractions;
+using System.Collections;
+using System;
+using YamlDotNet.RepresentationModel;
+using CsvHelper;
+using System.Linq;
+using System.Globalization;
+using System.Text;
+
+namespace Pretzel.Logic.Templating.Context
+{
+    public class Data : Drop
+    {
+        private readonly IFileSystem fileSystem;
+        private readonly string dataDirectory;
+        private System.Lazy<object> cachedResult;
+
+        public Data(IFileSystem fileSystem, string dataDirectory)
+        {
+            this.fileSystem = fileSystem;
+            this.dataDirectory = dataDirectory;
+        }
+
+        public override object this[object method]
+        {
+            get
+            {
+                var res = base[method];
+                if (res != null)
+                {
+                    return res;
+                }
+
+                if (cachedResult == null)
```

Comment:
The cache isn't dependant from `method`?
Isn't that a risk to get the same value for different data?

Reply:
No there isn't: `method` is called recursive:
```txt
site._data.people -> _data\people.yml -> SiteContext.Data(_data).Lazy(people.yml)
site._data.members.users -> _data\members\users.yml -> SiteContect.Data(_data).Lazy<Data(_data\members)>.Lazy(users.yml)
```
There is only one limitation though. Currently it's not possible to have a `people.yml` and a `_data\people` subfolder cause I don't know if another nested accessor will be called (i never get the full syntax with the dot's `_site.data.people.users` I only get one portion at the time `people` `users`.

Reply:
Uh I get you know. Yes it should probably a `Dictionary<method, Lazy>`.

Reply:
Fixed. Can you have another look?

Reply:
👍